docker_daemon/check.py

```diff
@@ -472,17 +473,42 @@ def _extract_tag_value(self, entity, tag_name):
 
         return entity["_tag_values"][tag_name]
 
+    def _is_ecs_agent_local(self):
+        """Return True if we can reach the ecs-agent over localhost, False otherwise.
+        This is needed because if the ecs-agent is started with --net=host it won't have an IP address attached.
+        """
+        if self.ecs_agent_local is not None:
+            return self.ecs_agent_local
+        result = ""
+        sock = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
+        sock.settimeout(5)
+        try:
+            result = sock.connect_ex(('localhost', ECS_INTROSPECT_DEFAULT_PORT))
+        except Exception as e:
+            self.log.debug("Unable to connect to ecs-agent. Exception: {0}".format(e))
+        sock.close()
+        if result == 0:
```

Comment:
if an exception was raised, result is undefined. To avoid that case you can do something like
```python
try:
    result = ...
except:
    self.log.debug(...)
    self.ecs_agent_local = False
else:
    if result == 0:
        ...
    else:
        ....
return self.ecs_agent_local
```